base/MAPL_LatLonGridFactory.F90

```diff
@@ -1387,60 +1389,96 @@ subroutine initialize_from_esmf_distGrid(this, dist_grid, lon_array, lat_array,
 
    end subroutine initialize_from_esmf_distGrid
 
+   logical function test_decomp_equals(this,a)
```

Comment:
Note you can still declare the result variable to be `result(equal)` to make the procedure read a bit better.  I only mention since you are changing the proc name anyway.